Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -543,22 +543,23 @@ public class HTTPClient {
             return failedTask
         }
 
-        let redirectHandler: RedirectHandler<Delegate.Response>?
-        switch self.configuration.redirectConfiguration.configuration {
-        case .follow(let max, let allowCycles):
+        let redirectHandler: RedirectHandler<Delegate.Response>? = {
             var request = request
             if request.redirectState == nil {
-                request.redirectState = .init(count: max, visited: allowCycles ? nil : Set())
+                request.redirectState = RedirectState(
+                    self.configuration.redirectConfiguration.configuration,
+                    initialURL: request.url.absoluteString
+                )
             }
-            redirectHandler = RedirectHandler<Delegate.Response>(request: request) { newRequest in
+            guard request.redirectState != nil else { return nil }
```

Comment:
I find this pattern pretty oblique: it's not obvious to me that having no redirect state means "don't follow redirects". Can we use a computed property on `request` to make it clearer?

Reply:
`redirectState` is now no longer a property on `request`. It never really belonged there in the first place and is now passed explicitly to the RedirectHandler. I also didn't like that `nil` means don't follow redirects but now that we explicitly forward it is a bit better.

If we still want to make it more explicit, I think we need to make `RedirectState` aware of the "don't follow redirects" case and ask it before we create a `RedirectHandler`.